AK/Math.h

```diff
@@ -996,14 +997,19 @@ constexpr T pow(T x, T y)
         return 0;
     if (y == 1)
         return x;
-    int y_as_int = (int)y;
-    if (y == (T)y_as_int) {
-        T result = x;
-        for (int i = 0; i < fabs<T>(y) - 1; ++i)
-            result *= x;
-        if (y < 0)
-            result = 1.0l / result;
-        return result;
+
+    // Take an integer fast path as long as the value fits within a 64-bit integer.
+    // To avoid an incorrect comparison due to rounding, we slightly adjust the 64-bit bounds towards zero.
+    if (y >= nextafter(static_cast<T>(NumericLimits<i64>::min()), Infinity<T>) && y <= nextafter(static_cast<T>(NumericLimits<i64>::max()), -Infinity<T>)) {
```

Comment:
What in the world? This blog post shows a similar logic working correctly without `nextafter` magic:

https://sillycross.github.io/2022/07/18/2022-07-18/#Pick-the-Right-Tool-Instead

Reply:
Well, it clearly fixed the UB, so I do not know. I would have preferred the normal solution too.

Reply:
That blog's magic values are what i am "synthesizing" here though...

Reply:
from earlier in the post

> While -2^63 (the smallest int64_t) has an exact representation in double, 2^63-1 (the largest int64_t) doesn’t. So we must be careful about the rounding problems when doing the comparison.

Reply:
Re-checked this, and if I change the lower bound to just the 64-bit min, our formatter says -9223372036854776000, which should actually be -9223372036854775808, which is the same as in the blogpost. For the upper bound, we have a value that is slightly off, but I my error was to use <= instead of <. With that, it should work with just static_cast'ing the min and max values, which I think is more readable than these huge constants.

Reply:
> our formatter says -9223372036854776000, which should actually be -9223372036854775808

And our formatter is completely correct because the value it outputs is precisely the shortest roundtrippable decimal representation for this double.